pkg/driver/driver_azure.go

```diff
@@ -359,48 +363,90 @@ func (d *AzureDriver) createVMNicDisk() (*compute.VirtualMachine, error) {
 	}
 	onARMAPISuccess(prometheusServiceSubnet, "subnet.Get")
 
+	/*
+		NIC creation
+	*/
+
 	// Creating NICParameters for new NIC creation request
 	NICParameters := d.getNICParameters(vmName, &subnet)
+
+	// NIC creation request
 	NICFuture, err := clients.nic.CreateOrUpdate(ctx, resourceGroupName, *NICParameters.Name, NICParameters)
 	if err != nil {
+		// Since machine creation failed, delete any infra resources created
+		deleteErr := clients.deleteVMNicDisk(ctx, resourceGroupName, vmName, nicName, diskName)
+		if deleteErr != nil {
+			glog.Errorf("Error occurred during resource clean up: %s", deleteErr)
```

Comment:
Should we return an error here?

Context:
While discussing offline with @prashanth26, we realized not returning the error could lead to an undesired situation of one extra machine created. Basically we try to delete all the resources of creation of any of them fail. If the deletion of the resources also fails, which is very likely, we may end up with orphan resources.
- We'd then expect Safety-controller to clean up, but personally I'd prefer to not rely on safety-net for features to work, it should only act while unknown behaviors from infra is observed.

_Also returning the error might not be the complete solution_, as deletion-flow would still delete the machine-object as the `ProviderID` field could be empty, this could again lead to orphan-resources.  

Overall we need to brainstorm a little for a better deletion logic for machines specifically for Azure, it can be tracked [here](https://github.com/gardener/machine-controller-manager/issues/350).

We don't have to convolute the purpose of this PR, we can solve it separately at the issue tagged [above](https://github.com/gardener/machine-controller-manager/issues/350).